app/services/braintree_services/payment_method_builder.rb

```diff
@@ -0,0 +1,43 @@
+module BraintreeServices
+  class PaymentMethodBuilder
+    def initialize(transaction:, customer: nil)
+      @transaction = transaction
+      @customer = customer
+    end
+
+    def create
+      Payment::Braintree::PaymentMethod.find_by(token: attributes['token']) ||
+        Payment::Braintree::PaymentMethod.create(attributes)
+    end
+
+    private
+
+    def attributes
+      case @transaction.payment_instrument_type
+      when Braintree::PaymentInstrumentType::CreditCard
+        credit_card_attributes
+      when Braintree::PaymentInstrumentType::PayPalAccount
+        paypal_attributes
+      end
+    end
+
+    def credit_card_attributes
```

Comment:
I'd go without ```send``` here and just compose the hash as in ```paypal_attributes```. 

IMHO:
* It will be easier to read, debug and understand
* It will have consistent hash keys (current version has strings/symbols mixed)
* It will be a bit faster and will construct fewer objects/calling fewer methods (not that this will matter much, I assume)